src/lib/models/inorder/Core.cc

```diff
@@ -116,14 +115,24 @@ bool Core::hasHalted() const {
   }
 
   // Core is considered to have halted when the fetch unit has halted, there
-  // are no uops at the head of any buffer, and no exception is currently being
-  // handled.
-  bool decodePending = fetchToDecodeBuffer_.getHeadSlots()[0].size() > 0;
-  bool executePending = decodeToExecuteBuffer_.getHeadSlots()[0] != nullptr;
-  bool writebackPending = completionSlots_[0].getHeadSlots()[0] != nullptr;
+  // are no uops pending in any buffer, the execute unit is not currently
+  // processing an instructions and no exception is currently being handled.
+
+  // Units place uops in buffer tail, but if buffer is stalled, uops do not
+  // move to head slots and so remain in tail slots. Buffer head appears
+  // empty. Check emptiness accordingly
+  auto decSlots = fetchToDecodeBuffer_.getPendingSlots()[0];
```

Comment:
There could be issues with this in the future regarding pipeline buffers of size 2 or more and the use of pipeline units which stall their buffers with uops still in the head slot. However, the inorder core is about to undergo some major changes, re complexity, and the logic in place to carry out the below check handled those above issues. Thus, we don't need to consider such issues for this PR.